src/pages/index.js

```diff
@@ -32,71 +87,145 @@ const cardPopOut = new PopupWithImage({
 
 cardPopOut.setEventListeners();
 
-function createCard(item) {
-  const card = new Card(item, "#localeCard", openModalImage);
-  return card.generateCard();
+const deleteCardModal = new PopupWithDelete({
+  popupSelector: ".modal_type_delete-image",
+  handleBtnClick: () => {
+    api.deleteCard(cardToDelete.data._id);
+    cardToDelete.deleteCard();
+    deleteCardModal.close();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error, the card should stay in the `DOM`.